// @grafana/grafana-foundation-sdk/dist/librarypanel/types.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultLibraryElementDTOMeta = exports.defaultLibraryElementDTOMetaUser = exports.defaultLibraryPanel = void 0;
const defaultLibraryPanel = () => ({
    uid: "",
    name: "",
    type: "",
    version: 0,
    model: {
        type: "",
        transparent: false,
        repeatDirection: "h",
    },
});
exports.defaultLibraryPanel = defaultLibraryPanel;
const defaultLibraryElementDTOMetaUser = () => ({
    id: 0,
    name: "",
    avatarUrl: "",
});
exports.defaultLibraryElementDTOMetaUser = defaultLibraryElementDTOMetaUser;
const defaultLibraryElementDTOMeta = () => ({
    folderName: "",
    folderUid: "",
    connectedDashboards: 0,
    created: "",
    updated: "",
    createdBy: (0, exports.defaultLibraryElementDTOMetaUser)(),
    updatedBy: (0, exports.defaultLibraryElementDTOMetaUser)(),
});
exports.defaultLibraryElementDTOMeta = defaultLibraryElementDTOMeta;
