// @grafana/grafana-foundation-sdk/dist/parca/dataqueryBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataqueryBuilder = void 0;
const tslib_1 = require("tslib");
const parca = tslib_1.__importStar(require("../parca"));
class DataqueryBuilder {
    constructor() {
        this.internal = parca.defaultDataquery();
    }
    build() {
        return this.internal;
    }
    labelSelector(labelSelector) {
        this.internal.labelSelector = labelSelector;
        return this;
    }
    profileTypeId(profileTypeId) {
        this.internal.profileTypeId = profileTypeId;
        return this;
    }
    refId(refId) {
        this.internal.refId = refId;
        return this;
    }
    hide(hide) {
        this.internal.hide = hide;
        return this;
    }
    queryType(queryType) {
        this.internal.queryType = queryType;
        return this;
    }
    datasource(datasource) {
        this.internal.datasource = datasource;
        return this;
    }
}
exports.DataqueryBuilder = DataqueryBuilder;
