// @grafana/grafana-foundation-sdk/dist/googlecloudmonitoring/timeSeriesQueryBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeSeriesQueryBuilder = void 0;
const tslib_1 = require("tslib");
const googlecloudmonitoring = tslib_1.__importStar(require("../googlecloudmonitoring"));
// Time Series sub-query properties.
class TimeSeriesQueryBuilder {
    constructor() {
        this.internal = googlecloudmonitoring.defaultTimeSeriesQuery();
    }
    build() {
        return this.internal;
    }
    // GCP project to execute the query against.
    projectName(projectName) {
        this.internal.projectName = projectName;
        return this;
    }
    // MQL query to be executed.
    query(query) {
        this.internal.query = query;
        return this;
    }
    // To disable the graphPeriod, it should explictly be set to 'disabled'.
    graphPeriod(graphPeriod) {
        this.internal.graphPeriod = graphPeriod;
        return this;
    }
}
exports.TimeSeriesQueryBuilder = TimeSeriesQueryBuilder;
