// @grafana/grafana-foundation-sdk/dist/cloudwatch/metricStatBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricStatBuilder = void 0;
const tslib_1 = require("tslib");
const cloudwatch = tslib_1.__importStar(require("../cloudwatch"));
class MetricStatBuilder {
    constructor() {
        this.internal = cloudwatch.defaultMetricStat();
    }
    build() {
        return this.internal;
    }
    // AWS region to query for the metric
    region(region) {
        this.internal.region = region;
        return this;
    }
    // A namespace is a container for CloudWatch metrics. Metrics in different namespaces are isolated from each other, so that metrics from different applications are not mistakenly aggregated into the same statistics. For example, Amazon EC2 uses the AWS/EC2 namespace.
    namespace(namespace) {
        this.internal.namespace = namespace;
        return this;
    }
    // Name of the metric
    metricName(metricName) {
        this.internal.metricName = metricName;
        return this;
    }
    // The dimensions of the metric
    dimensions(dimensions) {
        this.internal.dimensions = dimensions;
        return this;
    }
    // Only show metrics that exactly match all defined dimension names.
    matchExact(matchExact) {
        this.internal.matchExact = matchExact;
        return this;
    }
    // The length of time associated with a specific Amazon CloudWatch statistic. Can be specified by a number of seconds, 'auto', or as a duration string e.g. '15m' being 15 minutes
    period(period) {
        this.internal.period = period;
        return this;
    }
    // The ID of the AWS account to query for the metric, specifying `all` will query all accounts that the monitoring account is permitted to query.
    accountId(accountId) {
        this.internal.accountId = accountId;
        return this;
    }
    // Metric data aggregations over specified periods of time. For detailed definitions of the statistics supported by CloudWatch, see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Statistics-definitions.html.
    statistic(statistic) {
        this.internal.statistic = statistic;
        return this;
    }
    // @deprecated use statistic
    statistics(statistics) {
        this.internal.statistics = statistics;
        return this;
    }
}
exports.MetricStatBuilder = MetricStatBuilder;
