// @grafana/grafana-foundation-sdk/dist/role/roleBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleBuilder = void 0;
const tslib_1 = require("tslib");
const role = tslib_1.__importStar(require("../role"));
class RoleBuilder {
    constructor() {
        this.internal = role.defaultRole();
    }
    build() {
        return this.internal;
    }
    // The role identifier `managed:builtins:editor:permissions`
    name(name) {
        this.internal.name = name;
        return this;
    }
    // Optional display
    displayName(displayName) {
        this.internal.displayName = displayName;
        return this;
    }
    // Name of the team.
    groupName(groupName) {
        this.internal.groupName = groupName;
        return this;
    }
    // Role description
    description(description) {
        this.internal.description = description;
        return this;
    }
    // Do not show this role
    hidden(hidden) {
        this.internal.hidden = hidden;
        return this;
    }
}
exports.RoleBuilder = RoleBuilder;
