// @grafana/grafana-foundation-sdk/dist/dashboard/adHocVariableBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdHocVariableBuilder = void 0;
const tslib_1 = require("tslib");
const dashboard = tslib_1.__importStar(require("../dashboard"));
// A variable is a placeholder for a value. You can use variables in metric queries and in panel titles.
class AdHocVariableBuilder {
    constructor(name) {
        this.internal = dashboard.defaultVariableModel();
        this.internal.name = name;
        this.internal.type = dashboard.VariableType.Adhoc;
    }
    build() {
        return this.internal;
    }
    // Name of variable
    name(name) {
        this.internal.name = name;
        return this;
    }
    // Optional display name
    label(label) {
        this.internal.label = label;
        return this;
    }
    // Visibility configuration for the variable
    hide(hide) {
        this.internal.hide = hide;
        return this;
    }
    // Description of variable. It can be defined but `null`.
    description(description) {
        this.internal.description = description;
        return this;
    }
    // Data source used to fetch values for a variable. It can be defined but `null`.
    datasource(datasource) {
        this.internal.datasource = datasource;
        return this;
    }
}
exports.AdHocVariableBuilder = AdHocVariableBuilder;
