// @grafana/grafana-foundation-sdk/dist/loki/dataqueryBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataqueryBuilder = void 0;
const tslib_1 = require("tslib");
const loki = tslib_1.__importStar(require("../loki"));
class DataqueryBuilder {
    constructor() {
        this.internal = loki.defaultDataquery();
    }
    build() {
        return this.internal;
    }
    expr(expr) {
        this.internal.expr = expr;
        return this;
    }
    legendFormat(legendFormat) {
        this.internal.legendFormat = legendFormat;
        return this;
    }
    maxLines(maxLines) {
        this.internal.maxLines = maxLines;
        return this;
    }
    resolution(resolution) {
        this.internal.resolution = resolution;
        return this;
    }
    editorMode(editorMode) {
        this.internal.editorMode = editorMode;
        return this;
    }
    range(range) {
        this.internal.range = range;
        return this;
    }
    instant(instant) {
        this.internal.instant = instant;
        return this;
    }
    step(step) {
        this.internal.step = step;
        return this;
    }
    refId(refId) {
        this.internal.refId = refId;
        return this;
    }
    hide(hide) {
        this.internal.hide = hide;
        return this;
    }
    queryType(queryType) {
        this.internal.queryType = queryType;
        return this;
    }
    datasource(datasource) {
        this.internal.datasource = datasource;
        return this;
    }
}
exports.DataqueryBuilder = DataqueryBuilder;
