// @grafana/grafana-foundation-sdk/dist/xychart/types.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultOptions = exports.defaultScatterSeriesConfig = exports.defaultFieldConfig = exports.defaultXYDimensionConfig = exports.defaultScatterShow = exports.ScatterShow = exports.defaultSeriesMapping = exports.SeriesMapping = void 0;
const tslib_1 = require("tslib");
const common = tslib_1.__importStar(require("../common"));
// Auto is "table" in the UI
var SeriesMapping;
(function (SeriesMapping) {
    SeriesMapping["Auto"] = "auto";
    SeriesMapping["Manual"] = "manual";
})(SeriesMapping || (exports.SeriesMapping = SeriesMapping = {}));
const defaultSeriesMapping = () => (SeriesMapping.Auto);
exports.defaultSeriesMapping = defaultSeriesMapping;
var ScatterShow;
(function (ScatterShow) {
    ScatterShow["Points"] = "points";
    ScatterShow["Lines"] = "lines";
    ScatterShow["PointsAndLines"] = "points+lines";
})(ScatterShow || (exports.ScatterShow = ScatterShow = {}));
const defaultScatterShow = () => (ScatterShow.Points);
exports.defaultScatterShow = defaultScatterShow;
const defaultXYDimensionConfig = () => ({
    frame: 0,
});
exports.defaultXYDimensionConfig = defaultXYDimensionConfig;
const defaultFieldConfig = () => ({
    show: ScatterShow.Points,
    label: common.VisibilityMode.Auto,
});
exports.defaultFieldConfig = defaultFieldConfig;
const defaultScatterSeriesConfig = () => ({
    show: ScatterShow.Points,
    label: common.VisibilityMode.Auto,
});
exports.defaultScatterSeriesConfig = defaultScatterSeriesConfig;
const defaultOptions = () => ({
    dims: (0, exports.defaultXYDimensionConfig)(),
    legend: common.defaultVizLegendOptions(),
    tooltip: common.defaultVizTooltipOptions(),
    series: [],
});
exports.defaultOptions = defaultOptions;
