// @grafana/grafana-foundation-sdk/dist/azuremonitor/resourceNamesQueryBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceNamesQueryBuilder = void 0;
const tslib_1 = require("tslib");
const azuremonitor = tslib_1.__importStar(require("../azuremonitor"));
class ResourceNamesQueryBuilder {
    constructor() {
        this.internal = azuremonitor.defaultResourceNamesQuery();
        this.internal.kind = "ResourceNamesQuery";
    }
    build() {
        return this.internal;
    }
    rawQuery(rawQuery) {
        this.internal.rawQuery = rawQuery;
        return this;
    }
    subscription(subscription) {
        this.internal.subscription = subscription;
        return this;
    }
    resourceGroup(resourceGroup) {
        this.internal.resourceGroup = resourceGroup;
        return this;
    }
    metricNamespace(metricNamespace) {
        this.internal.metricNamespace = metricNamespace;
        return this;
    }
}
exports.ResourceNamesQueryBuilder = ResourceNamesQueryBuilder;
