// @grafana/grafana-foundation-sdk/dist/accesspolicy/accessPolicyBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessPolicyBuilder = void 0;
const tslib_1 = require("tslib");
const accesspolicy = tslib_1.__importStar(require("../accesspolicy"));
class AccessPolicyBuilder {
    constructor() {
        this.internal = accesspolicy.defaultAccessPolicy();
    }
    build() {
        return this.internal;
    }
    // The scope where these policies should apply
    scope(scope) {
        const scopeResource = scope.build();
        this.internal.scope = scopeResource;
        return this;
    }
    // The role that must apply this policy
    role(role) {
        const roleResource = role.build();
        this.internal.role = roleResource;
        return this;
    }
    // The set of rules to apply.  Note that * is required to modify
    // access policy rules, and that "none" will reject all actions
    rules(rules) {
        if (!this.internal.rules) {
            this.internal.rules = [];
        }
        const rulesResource = rules.build();
        this.internal.rules.push(rulesResource);
        return this;
    }
}
exports.AccessPolicyBuilder = AccessPolicyBuilder;
