// @grafana/grafana-foundation-sdk/dist/testdata/scenarioBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScenarioBuilder = void 0;
const tslib_1 = require("tslib");
const testdata = tslib_1.__importStar(require("../testdata"));
// TODO: Should this live here given it's not used in the dataquery?
class ScenarioBuilder {
    constructor() {
        this.internal = testdata.defaultScenario();
    }
    build() {
        return this.internal;
    }
    id(id) {
        this.internal.id = id;
        return this;
    }
    name(name) {
        this.internal.name = name;
        return this;
    }
    stringInput(stringInput) {
        this.internal.stringInput = stringInput;
        return this;
    }
    description(description) {
        this.internal.description = description;
        return this;
    }
    hideAliasField(hideAliasField) {
        this.internal.hideAliasField = hideAliasField;
        return this;
    }
}
exports.ScenarioBuilder = ScenarioBuilder;
