// @grafana/grafana-foundation-sdk/dist/elasticsearch/dataqueryBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataqueryBuilder = void 0;
const tslib_1 = require("tslib");
const elasticsearch = tslib_1.__importStar(require("../elasticsearch"));
class DataqueryBuilder {
    constructor() {
        this.internal = elasticsearch.defaultDataquery();
    }
    build() {
        return this.internal;
    }
    alias(alias) {
        this.internal.alias = alias;
        return this;
    }
    query(query) {
        this.internal.query = query;
        return this;
    }
    timeField(timeField) {
        this.internal.timeField = timeField;
        return this;
    }
    bucketAggs(bucketAggs) {
        this.internal.bucketAggs = bucketAggs;
        return this;
    }
    metrics(metrics) {
        this.internal.metrics = metrics;
        return this;
    }
    refId(refId) {
        this.internal.refId = refId;
        return this;
    }
    hide(hide) {
        this.internal.hide = hide;
        return this;
    }
    queryType(queryType) {
        this.internal.queryType = queryType;
        return this;
    }
    datasource(datasource) {
        this.internal.datasource = datasource;
        return this;
    }
}
exports.DataqueryBuilder = DataqueryBuilder;
