// @grafana/grafana-foundation-sdk/dist/common/dataSourceJsonDataBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataSourceJsonDataBuilder = void 0;
const tslib_1 = require("tslib");
const common = tslib_1.__importStar(require("../common"));
// TODO docs
class DataSourceJsonDataBuilder {
    constructor() {
        this.internal = common.defaultDataSourceJsonData();
    }
    build() {
        return this.internal;
    }
    authType(authType) {
        this.internal.authType = authType;
        return this;
    }
    defaultRegion(defaultRegion) {
        this.internal.defaultRegion = defaultRegion;
        return this;
    }
    profile(profile) {
        this.internal.profile = profile;
        return this;
    }
    manageAlerts(manageAlerts) {
        this.internal.manageAlerts = manageAlerts;
        return this;
    }
    alertmanagerUid(alertmanagerUid) {
        this.internal.alertmanagerUid = alertmanagerUid;
        return this;
    }
}
exports.DataSourceJsonDataBuilder = DataSourceJsonDataBuilder;
