// @grafana/grafana-foundation-sdk/dist/common/mapLayerOptionsBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MapLayerOptionsBuilder = void 0;
const tslib_1 = require("tslib");
const common = tslib_1.__importStar(require("../common"));
class MapLayerOptionsBuilder {
    constructor() {
        this.internal = common.defaultMapLayerOptions();
    }
    build() {
        return this.internal;
    }
    type(type) {
        this.internal.type = type;
        return this;
    }
    // configured unique display name
    name(name) {
        this.internal.name = name;
        return this;
    }
    // Custom options depending on the type
    config(config) {
        this.internal.config = config;
        return this;
    }
    // Common method to define geometry fields
    location(location) {
        const locationResource = location.build();
        this.internal.location = locationResource;
        return this;
    }
    // Defines a frame MatcherConfig that may filter data for the given layer
    filterData(filterData) {
        this.internal.filterData = filterData;
        return this;
    }
    // Common properties:
    // https://openlayers.org/en/latest/apidoc/module-ol_layer_Base-BaseLayer.html
    // Layer opacity (0-1)
    opacity(opacity) {
        this.internal.opacity = opacity;
        return this;
    }
    // Check tooltip (defaults to true)
    tooltip(tooltip) {
        this.internal.tooltip = tooltip;
        return this;
    }
}
exports.MapLayerOptionsBuilder = MapLayerOptionsBuilder;
