// @grafana/grafana-foundation-sdk/dist/common/reduceDataOptionsBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReduceDataOptionsBuilder = void 0;
const tslib_1 = require("tslib");
const common = tslib_1.__importStar(require("../common"));
// TODO docs
class ReduceDataOptionsBuilder {
    constructor() {
        this.internal = common.defaultReduceDataOptions();
    }
    build() {
        return this.internal;
    }
    // If true show each row value
    values(values) {
        this.internal.values = values;
        return this;
    }
    // if showing all values limit
    limit(limit) {
        this.internal.limit = limit;
        return this;
    }
    // When !values, pick one value for the whole field
    calcs(calcs) {
        this.internal.calcs = calcs;
        return this;
    }
    // Which fields to show.  By default this is only numeric fields
    fields(fields) {
        this.internal.fields = fields;
        return this;
    }
}
exports.ReduceDataOptionsBuilder = ReduceDataOptionsBuilder;
