// @grafana/grafana-foundation-sdk/dist/common/vizLegendOptionsBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VizLegendOptionsBuilder = void 0;
const tslib_1 = require("tslib");
const common = tslib_1.__importStar(require("../common"));
// TODO docs
class VizLegendOptionsBuilder {
    constructor() {
        this.internal = common.defaultVizLegendOptions();
    }
    build() {
        return this.internal;
    }
    displayMode(displayMode) {
        this.internal.displayMode = displayMode;
        return this;
    }
    placement(placement) {
        this.internal.placement = placement;
        return this;
    }
    showLegend(showLegend) {
        this.internal.showLegend = showLegend;
        return this;
    }
    asTable(asTable) {
        this.internal.asTable = asTable;
        return this;
    }
    isVisible(isVisible) {
        this.internal.isVisible = isVisible;
        return this;
    }
    sortBy(sortBy) {
        this.internal.sortBy = sortBy;
        return this;
    }
    sortDesc(sortDesc) {
        this.internal.sortDesc = sortDesc;
        return this;
    }
    width(width) {
        this.internal.width = width;
        return this;
    }
    calcs(calcs) {
        this.internal.calcs = calcs;
        return this;
    }
}
exports.VizLegendOptionsBuilder = VizLegendOptionsBuilder;
