// @grafana/grafana-foundation-sdk/dist/azuremonitor/azureLogsQueryBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureLogsQueryBuilder = void 0;
const tslib_1 = require("tslib");
const azuremonitor = tslib_1.__importStar(require("../azuremonitor"));
// Azure Monitor Logs sub-query properties
class AzureLogsQueryBuilder {
    constructor() {
        this.internal = azuremonitor.defaultAzureLogsQuery();
    }
    build() {
        return this.internal;
    }
    // KQL query to be executed.
    query(query) {
        this.internal.query = query;
        return this;
    }
    // Specifies the format results should be returned as.
    resultFormat(resultFormat) {
        this.internal.resultFormat = resultFormat;
        return this;
    }
    // Array of resource URIs to be queried.
    resources(resources) {
        this.internal.resources = resources;
        return this;
    }
    // If set to true the dashboard time range will be used as a filter for the query. Otherwise the query time ranges will be used. Defaults to false.
    dashboardTime(dashboardTime) {
        this.internal.dashboardTime = dashboardTime;
        return this;
    }
    // If dashboardTime is set to true this value dictates which column the time filter will be applied to. Defaults to the first tables timeSpan column, the first datetime column found, or TimeGenerated
    timeColumn(timeColumn) {
        this.internal.timeColumn = timeColumn;
        return this;
    }
    // Workspace ID. This was removed in Grafana 8, but remains for backwards compat.
    workspace(workspace) {
        this.internal.workspace = workspace;
        return this;
    }
    // @deprecated Use resources instead
    resource(resource) {
        this.internal.resource = resource;
        return this;
    }
    // @deprecated Use dashboardTime instead
    intersectTime(intersectTime) {
        this.internal.intersectTime = intersectTime;
        return this;
    }
}
exports.AzureLogsQueryBuilder = AzureLogsQueryBuilder;
