// @grafana/grafana-foundation-sdk/dist/azuremonitor/metricDefinitionsQueryBuilder.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricDefinitionsQueryBuilder = void 0;
const tslib_1 = require("tslib");
const azuremonitor = tslib_1.__importStar(require("../azuremonitor"));
// @deprecated Use MetricNamespaceQuery instead
class MetricDefinitionsQueryBuilder {
    constructor() {
        this.internal = azuremonitor.defaultMetricDefinitionsQuery();
        this.internal.kind = "MetricDefinitionsQuery";
    }
    build() {
        return this.internal;
    }
    rawQuery(rawQuery) {
        this.internal.rawQuery = rawQuery;
        return this;
    }
    subscription(subscription) {
        this.internal.subscription = subscription;
        return this;
    }
    resourceGroup(resourceGroup) {
        this.internal.resourceGroup = resourceGroup;
        return this;
    }
    metricNamespace(metricNamespace) {
        this.internal.metricNamespace = metricNamespace;
        return this;
    }
    resourceName(resourceName) {
        this.internal.resourceName = resourceName;
        return this;
    }
}
exports.MetricDefinitionsQueryBuilder = MetricDefinitionsQueryBuilder;
