// @grafana/grafana-foundation-sdk/dist/elasticsearch/types.gen.js
"use strict";
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultBaseMovingAverageModelSettings = exports.defaultMovingAverageModelOption = exports.defaultMovingAverageModel = exports.MovingAverageModel = exports.defaultPipelineMetricAggregationWithMultipleBucketPaths = exports.defaultBasePipelineMetricAggregation = exports.defaultRate = exports.defaultLogs = exports.defaultRawData = exports.defaultRawDocument = exports.defaultUniqueCount = exports.defaultPercentiles = exports.defaultExtendedStats = exports.defaultExtendedStat = exports.defaultExtendedStatMetaType = exports.ExtendedStatMetaType = exports.defaultMin = exports.defaultMax = exports.defaultSum = exports.defaultAverage = exports.defaultCount = exports.defaultMetricAggregationWithInlineScript = exports.defaultInlineScript = exports.defaultMetricAggregationWithMissingSupport = exports.defaultMetricAggregationWithField = exports.defaultPipelineVariable = exports.defaultBaseMetricAggregation = exports.defaultMetricAggregationType = exports.defaultPipelineMetricAggregationType = exports.PipelineMetricAggregationType = exports.defaultGeoHashGridSettings = exports.defaultGeoHashGrid = exports.defaultFiltersSettings = exports.defaultFilter = exports.defaultFilters = exports.defaultTermsSettings = exports.defaultTerms = exports.defaultNested = exports.defaultTermsOrder = exports.TermsOrder = exports.defaultHistogramSettings = exports.defaultHistogram = exports.defaultDateHistogramSettings = exports.defaultDateHistogram = exports.defaultBucketAggregationWithField = exports.defaultBaseBucketAggregation = exports.defaultBucketAggregationType = exports.BucketAggregationType = exports.defaultMetricAggregation = exports.defaultBucketAggregation = void 0;
exports.defaultDataquery = exports.defaultMetricAggregationWithSettings = exports.defaultPipelineMetricAggregation = exports.defaultTopMetrics = exports.defaultBucketScript = exports.defaultCumulativeSum = exports.defaultSerialDiff = exports.defaultDerivative = exports.defaultMovingFunction = exports.defaultMovingAverage = exports.defaultMovingAverageHoltWintersModelSettings = exports.defaultMovingAverageHoltModelSettings = exports.defaultMovingAverageEWMAModelSettings = exports.defaultMovingAverageLinearModelSettings = exports.defaultMovingAverageSimpleModelSettings = void 0;
const defaultBucketAggregation = () => ((0, exports.defaultDateHistogram)());
exports.defaultBucketAggregation = defaultBucketAggregation;
const defaultMetricAggregation = () => ((0, exports.defaultCount)());
exports.defaultMetricAggregation = defaultMetricAggregation;
var BucketAggregationType;
(function (BucketAggregationType) {
    BucketAggregationType["Terms"] = "terms";
    BucketAggregationType["Filters"] = "filters";
    BucketAggregationType["GeohashGrid"] = "geohash_grid";
    BucketAggregationType["DateHistogram"] = "date_histogram";
    BucketAggregationType["Histogram"] = "histogram";
    BucketAggregationType["Nested"] = "nested";
})(BucketAggregationType || (exports.BucketAggregationType = BucketAggregationType = {}));
const defaultBucketAggregationType = () => (BucketAggregationType.Terms);
exports.defaultBucketAggregationType = defaultBucketAggregationType;
const defaultBaseBucketAggregation = () => ({
    id: "",
    type: BucketAggregationType.Terms,
});
exports.defaultBaseBucketAggregation = defaultBaseBucketAggregation;
const defaultBucketAggregationWithField = () => ({
    id: "",
    type: BucketAggregationType.Terms,
});
exports.defaultBucketAggregationWithField = defaultBucketAggregationWithField;
const defaultDateHistogram = () => ({
    id: "",
    type: "date_histogram",
});
exports.defaultDateHistogram = defaultDateHistogram;
const defaultDateHistogramSettings = () => ({});
exports.defaultDateHistogramSettings = defaultDateHistogramSettings;
const defaultHistogram = () => ({
    id: "",
    type: "histogram",
});
exports.defaultHistogram = defaultHistogram;
const defaultHistogramSettings = () => ({});
exports.defaultHistogramSettings = defaultHistogramSettings;
var TermsOrder;
(function (TermsOrder) {
    TermsOrder["Desc"] = "desc";
    TermsOrder["Asc"] = "asc";
})(TermsOrder || (exports.TermsOrder = TermsOrder = {}));
const defaultTermsOrder = () => (TermsOrder.Desc);
exports.defaultTermsOrder = defaultTermsOrder;
const defaultNested = () => ({
    id: "",
    type: "nested",
});
exports.defaultNested = defaultNested;
const defaultTerms = () => ({
    id: "",
    type: "terms",
});
exports.defaultTerms = defaultTerms;
const defaultTermsSettings = () => ({});
exports.defaultTermsSettings = defaultTermsSettings;
const defaultFilters = () => ({
    id: "",
    type: "filters",
});
exports.defaultFilters = defaultFilters;
const defaultFilter = () => ({
    query: "",
    label: "",
});
exports.defaultFilter = defaultFilter;
const defaultFiltersSettings = () => ({});
exports.defaultFiltersSettings = defaultFiltersSettings;
const defaultGeoHashGrid = () => ({
    id: "",
    type: "geohash_grid",
});
exports.defaultGeoHashGrid = defaultGeoHashGrid;
const defaultGeoHashGridSettings = () => ({});
exports.defaultGeoHashGridSettings = defaultGeoHashGridSettings;
var PipelineMetricAggregationType;
(function (PipelineMetricAggregationType) {
    PipelineMetricAggregationType["MovingAvg"] = "moving_avg";
    PipelineMetricAggregationType["MovingFn"] = "moving_fn";
    PipelineMetricAggregationType["Derivative"] = "derivative";
    PipelineMetricAggregationType["SerialDiff"] = "serial_diff";
    PipelineMetricAggregationType["CumulativeSum"] = "cumulative_sum";
    PipelineMetricAggregationType["BucketScript"] = "bucket_script";
})(PipelineMetricAggregationType || (exports.PipelineMetricAggregationType = PipelineMetricAggregationType = {}));
const defaultPipelineMetricAggregationType = () => (PipelineMetricAggregationType.MovingAvg);
exports.defaultPipelineMetricAggregationType = defaultPipelineMetricAggregationType;
const defaultMetricAggregationType = () => ("count");
exports.defaultMetricAggregationType = defaultMetricAggregationType;
const defaultBaseMetricAggregation = () => ({
    type: (0, exports.defaultMetricAggregationType)(),
    id: "",
});
exports.defaultBaseMetricAggregation = defaultBaseMetricAggregation;
const defaultPipelineVariable = () => ({
    name: "",
    pipelineAgg: "",
});
exports.defaultPipelineVariable = defaultPipelineVariable;
const defaultMetricAggregationWithField = () => ({
    type: (0, exports.defaultMetricAggregationType)(),
    id: "",
});
exports.defaultMetricAggregationWithField = defaultMetricAggregationWithField;
const defaultMetricAggregationWithMissingSupport = () => ({
    type: (0, exports.defaultMetricAggregationType)(),
    id: "",
});
exports.defaultMetricAggregationWithMissingSupport = defaultMetricAggregationWithMissingSupport;
const defaultInlineScript = () => ("");
exports.defaultInlineScript = defaultInlineScript;
const defaultMetricAggregationWithInlineScript = () => ({
    type: (0, exports.defaultMetricAggregationType)(),
    id: "",
});
exports.defaultMetricAggregationWithInlineScript = defaultMetricAggregationWithInlineScript;
const defaultCount = () => ({
    type: "count",
    id: "",
});
exports.defaultCount = defaultCount;
const defaultAverage = () => ({
    type: "avg",
    id: "",
});
exports.defaultAverage = defaultAverage;
const defaultSum = () => ({
    type: "sum",
    id: "",
});
exports.defaultSum = defaultSum;
const defaultMax = () => ({
    type: "max",
    id: "",
});
exports.defaultMax = defaultMax;
const defaultMin = () => ({
    type: "min",
    id: "",
});
exports.defaultMin = defaultMin;
var ExtendedStatMetaType;
(function (ExtendedStatMetaType) {
    ExtendedStatMetaType["Avg"] = "avg";
    ExtendedStatMetaType["Min"] = "min";
    ExtendedStatMetaType["Max"] = "max";
    ExtendedStatMetaType["Sum"] = "sum";
    ExtendedStatMetaType["Count"] = "count";
    ExtendedStatMetaType["StdDeviation"] = "std_deviation";
    ExtendedStatMetaType["StdDeviationBoundsUpper"] = "std_deviation_bounds_upper";
    ExtendedStatMetaType["StdDeviationBoundsLower"] = "std_deviation_bounds_lower";
})(ExtendedStatMetaType || (exports.ExtendedStatMetaType = ExtendedStatMetaType = {}));
const defaultExtendedStatMetaType = () => (ExtendedStatMetaType.Avg);
exports.defaultExtendedStatMetaType = defaultExtendedStatMetaType;
const defaultExtendedStat = () => ({
    label: "",
    value: ExtendedStatMetaType.Avg,
});
exports.defaultExtendedStat = defaultExtendedStat;
const defaultExtendedStats = () => ({
    type: "extended_stats",
    id: "",
});
exports.defaultExtendedStats = defaultExtendedStats;
const defaultPercentiles = () => ({
    type: "percentiles",
    id: "",
});
exports.defaultPercentiles = defaultPercentiles;
const defaultUniqueCount = () => ({
    type: "cardinality",
    id: "",
});
exports.defaultUniqueCount = defaultUniqueCount;
const defaultRawDocument = () => ({
    type: "raw_document",
    id: "",
});
exports.defaultRawDocument = defaultRawDocument;
const defaultRawData = () => ({
    type: "raw_data",
    id: "",
});
exports.defaultRawData = defaultRawData;
const defaultLogs = () => ({
    type: "logs",
    id: "",
});
exports.defaultLogs = defaultLogs;
const defaultRate = () => ({
    type: "rate",
    id: "",
});
exports.defaultRate = defaultRate;
const defaultBasePipelineMetricAggregation = () => ({
    type: "",
    id: "",
});
exports.defaultBasePipelineMetricAggregation = defaultBasePipelineMetricAggregation;
const defaultPipelineMetricAggregationWithMultipleBucketPaths = () => ({
    type: (0, exports.defaultMetricAggregationType)(),
    id: "",
});
exports.defaultPipelineMetricAggregationWithMultipleBucketPaths = defaultPipelineMetricAggregationWithMultipleBucketPaths;
var MovingAverageModel;
(function (MovingAverageModel) {
    MovingAverageModel["Simple"] = "simple";
    MovingAverageModel["Linear"] = "linear";
    MovingAverageModel["Ewma"] = "ewma";
    MovingAverageModel["Holt"] = "holt";
    MovingAverageModel["HoltWinters"] = "holt_winters";
})(MovingAverageModel || (exports.MovingAverageModel = MovingAverageModel = {}));
const defaultMovingAverageModel = () => (MovingAverageModel.Simple);
exports.defaultMovingAverageModel = defaultMovingAverageModel;
const defaultMovingAverageModelOption = () => ({
    label: "",
    value: MovingAverageModel.Simple,
});
exports.defaultMovingAverageModelOption = defaultMovingAverageModelOption;
const defaultBaseMovingAverageModelSettings = () => ({
    model: MovingAverageModel.Simple,
    window: "",
    predict: "",
});
exports.defaultBaseMovingAverageModelSettings = defaultBaseMovingAverageModelSettings;
const defaultMovingAverageSimpleModelSettings = () => ({
    model: "simple",
    window: "",
    predict: "",
});
exports.defaultMovingAverageSimpleModelSettings = defaultMovingAverageSimpleModelSettings;
const defaultMovingAverageLinearModelSettings = () => ({
    model: "linear",
    window: "",
    predict: "",
});
exports.defaultMovingAverageLinearModelSettings = defaultMovingAverageLinearModelSettings;
const defaultMovingAverageEWMAModelSettings = () => ({
    model: "ewma",
    window: "",
    minimize: false,
    predict: "",
});
exports.defaultMovingAverageEWMAModelSettings = defaultMovingAverageEWMAModelSettings;
const defaultMovingAverageHoltModelSettings = () => ({
    model: "holt",
    settings: {},
    window: "",
    minimize: false,
    predict: "",
});
exports.defaultMovingAverageHoltModelSettings = defaultMovingAverageHoltModelSettings;
const defaultMovingAverageHoltWintersModelSettings = () => ({
    model: "holt_winters",
    settings: {},
    window: "",
    minimize: false,
    predict: "",
});
exports.defaultMovingAverageHoltWintersModelSettings = defaultMovingAverageHoltWintersModelSettings;
const defaultMovingAverage = () => ({
    type: "moving_avg",
    id: "",
});
exports.defaultMovingAverage = defaultMovingAverage;
const defaultMovingFunction = () => ({
    type: "moving_fn",
    id: "",
});
exports.defaultMovingFunction = defaultMovingFunction;
const defaultDerivative = () => ({
    type: "derivative",
    id: "",
});
exports.defaultDerivative = defaultDerivative;
const defaultSerialDiff = () => ({
    type: "serial_diff",
    id: "",
});
exports.defaultSerialDiff = defaultSerialDiff;
const defaultCumulativeSum = () => ({
    type: "cumulative_sum",
    id: "",
});
exports.defaultCumulativeSum = defaultCumulativeSum;
const defaultBucketScript = () => ({
    type: "bucket_script",
    id: "",
});
exports.defaultBucketScript = defaultBucketScript;
const defaultTopMetrics = () => ({
    type: "top_metrics",
    id: "",
});
exports.defaultTopMetrics = defaultTopMetrics;
const defaultPipelineMetricAggregation = () => ((0, exports.defaultMovingAverage)());
exports.defaultPipelineMetricAggregation = defaultPipelineMetricAggregation;
const defaultMetricAggregationWithSettings = () => ((0, exports.defaultBucketScript)());
exports.defaultMetricAggregationWithSettings = defaultMetricAggregationWithSettings;
const defaultDataquery = () => ({
    _implementsDataqueryVariant: () => { },
});
exports.defaultDataquery = defaultDataquery;
